### PR TITLE
made OpenFeature implement EvaluationLifeCycle

### PR DESCRIPTION
**Pull request by anudeepreddy**
Signed-off-by: Anudeep Reddy <anudeepc85@gmail.com>

Hello👋,

I have worked on #13, the following are the changes that are included in the PR:
- Made the OpenFeature class implement EvaluationLifeCycle.
- made the getInstance static method public.
- changes the remaining static methods to non-static.

I would appreciate any feedback.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class OpenFeatureClient implements Client {
   readonly metadata: ClientMetadata;
   readonly context: EvaluationContext;
   private _hooks: Hook[] = [];
-
+  private openFeature: OpenFeature;
```
Comment (toddbaert) [inline src/client.ts:26]:
```suggestion

  private openFeature: OpenFeature;
```

File: src/open-feature.ts [hunk 2]
```diff
@@ -11,15 +11,15 @@ type OpenFeatureGlobal = {
 const _global = global as OpenFeatureGlobal;
 
 // TODO: make implement EvaluationLifeCycle, but statically...
-export class OpenFeature {
+export class OpenFeature implements EvaluationLifeCycle {
```
Comment (toddbaert) [inline src/open-feature.ts:14]: Don't export this class at all, instead, just add something like this at the bottom of the file:

```
export const openFeature = OpenFeature.getInstance();
```

This way, we are EXPORTING an instance, and nobody should (nobody will be able) to call `OpenFeature.getInstance`.

Then, consuming code will just do:

`import { openFeature } from './open-feature.js'`, which will already be a singleton!

**Review by toddbaert — CHANGES_REQUESTED** [review]
Great work so far! Please just export a singleton instead of the class though.

**Comment by toddbaert** [comment]
@weyert This change will be breaking only in that it exports a lower case `openFeature` (an instance) instead of `OpenFeature` (a class). This is mostly a style choice, but I think it's the correct one and now is the time to do it. You should only have to change the case, nothing else should break.

**Comment by weyert** [comment]
Why not rename the class to something different so that it doesn’t conflict with `OpenFeature`? To me, capitalised name indicates most often it’s a global/singleton instance

**Comment by toddbaert** [comment]
> Why not rename the class to something different so that it doesn’t conflict with `OpenFeature`? To me, capitalised name indicates most often it’s a global/singleton instance

We no longer export the `OpenFeature` class, we just export this instance, which is better I think. It was supposed to be used as a singleton, so just exporting one is better.

I don't really mind if we export `OpenFeature` vs `openFeature`. I've seen both in singletons, TBH.

**Comment by anudeepreddy** [comment]
> I don't really mind if we export OpenFeature vs openFeature. I've seen both in singletons, TBH.

What should be the name of the class if we export the singleton as OpenFeature? In case if Java and python it is named OpenFeatureAPI, should the same be done here aswell?

**Comment by toddbaert** [comment]
> What should be the name of the class if we export the singleton as OpenFeature? In case if Java and python it is named OpenFeatureAPI, should the same be done here as well?

Yep, I think that's fine.

**Review by toddbaert — APPROVED** [review]
LGTM. @weyert I think this new version will have zero breaking changes.

**Comment by anudeepreddy** [comment]
Looking forward to making more contributions 😄

**Comment by beeme1mr** [comment]
Thanks @anudeepreddy!

**Pull request merged**